clang/lib/3C/ProgramInfo.cpp

```diff
@@ -364,11 +369,11 @@ bool ProgramInfo::link() {
         std::string Rsn =
             "Unchecked pointer in parameter or return of static function " +
             FuncName + " in " + FileName;
-        if (!G->getReturnVar()->getIsGeneric())
-          G->getReturnVar()->constrainToWild(CS, Rsn);
+        if (!G->getExternalReturn()->getIsGeneric())
+          G->getExternalReturn()->constrainToWild(CS, Rsn);
```

Comment:
Why only the external here, but both internal and external for the `extern` function case, above?